data-prepper-api/src/main/java/org/opensearch/dataprepper/logging/DataPrepperMarkers.java

```diff
@@ -0,0 +1,15 @@
+package org.opensearch.dataprepper.logging;
+
+import org.slf4j.Marker;
+import org.slf4j.MarkerFactory;
+
+public final class DataPrepperMarkers {
+    public static final Marker EVENT_MARKER = MarkerFactory.getMarker("EVENT");
+    public static final Marker SENSITIVE_MARKER = MarkerFactory.getMarker("SENSITIVE");
+
+    static {
+        EVENT_MARKER.add(SENSITIVE_MARKER);
```

Comment:
From the documentation, this is adding a reference, which seems similar to adding a child.

So I think you are making sensitive a child of event here.

Reply:
Yes, it is documented here https://logging.apache.org/log4j/2.x/manual/filters.html#MarkerFilter

```
A match occurs when the Marker name matches either the Log Event's Marker or one of its parents.
```

So matching sensitive will also match event while matching event will leave out sensitive.

Reply:
I think from what I read from this 

> A match occurs when the Marker name matches either the Log Event's Marker or one of its parents.

I think matching EVENT should match both EVENT and SENSITIVE because SENSITIVE is a child of EVENT.
@chenqi0805 can you clarify this?
 
 Also, as you mentioned in PR description it doesn't look right. How does matching just SENSITIVE matches it's parent (EVENT)?

Reply:
Discussed offline: filter picks up marker configured in log4j configuration and compares it against those markers appearing in log event